Add knowledge checks and links to Introduction to RSpec lesson (#23157)
TheOdinProject/curriculum: ruby_programming/testing_with_rspec/introduction_to_rspec.md

- Briefly look over the RSpec styling and syntax recommended by [BetterSpecs](http://www.betterspecs.org/) and read through the first six guidelines.
- The [RSpec Cheat Sheet](https://www.dropbox.com/s/mij3d0bhi29outc/rspec_cheatsheet_attributed.pdf) should help you avoid Googling every new bit of syntax.
- Solidify these concepts with a [shameless plug](https://medium.com/@mindovermiles262/getting-started-with-rspec-part-1-9418909f5e53) from another Odin Project contributor.

### Knowledge Checks 
This section contains questions for you to check your understanding of this lesson. If you’re having trouble answering the questions below on your own, review the material above to find the answer.

> - <a class='knowledge-check-link' href='#tests-folder'>In which folder do we keep our test files?</a>

Reviewer: ```suggestion
- <a class='knowledge-check-link' href='#tests-folder'>What do you name the folder that contains your test files?</a>
```

- <a class='knowledge-check-link' href='#keyword-describe'>What does the keyword `describe` define?</a>
- <a class='knowledge-check-link' href='#keyword-it'>What does the keyword `it` define?</a>

> - <a class='knowledge-check-link' href='#expect-clauses'>How many expect clauses can we write for one test case?</a>

Reviewer: ```suggestion
- <a class='knowledge-check-link' href='#expect-clauses'>How many expect clauses can you write for one test case?</a>
```
We want to speak in the second person in Knowledge Checks whenever it makes sense to do so.
